### DISCUSSION TITLE
Barrier function material in DerivativeMultiPhaseMaterial

### DISCUSSION THREAD

**Discussion by LiWen-HUST** [idaholab/moose#16193] [Q&A Modules: Phase field] [answered]
Hello all,
I want to use more than one barrier function material in my PhaseField model. Like in the paper "Formation path of δ hydrides in zirconium by multiphase field modeling" the authors used three different barrier function materials, "MultiBarrierFunctionMaterial", "AsymmetricCrossTermBarrierFunctionMaterial" and "ThirdPhaseSuppressionMaterial". However in the "DerivativeMultiPhaseMaterial" material, only one barrier function "g" and one prefactor "W" can be used. How can I use more than one barrier function in PhaseField module?

BEST
Li

**Comment by souravmat-git** [comment, marked as answer]
Hello,

Check the following materials to create the barrier function used in the paper mentioned by you. 

moose/modules/phase_field/src/materials/AsymmetricCrossTermBarrierFunctionMaterial.C
moose/modules/phase_field/src/materials/ThirdPhaseSuppressionMaterial.C

Also check the following input file:
moose/modules/phase_field/test/tests/MultiPhase/asymmetriccrosstermbarrierfunction.i
